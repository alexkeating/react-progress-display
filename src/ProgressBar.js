/**
 * Created by alexkeating on 6/6/17.
 */

// What will this progress bar do?
// 0.1.0 should be a straight line that can be partitioned.

import React from 'react';

const ProgressBar = () => {
  return (
    <div>
      <h1>Hi</h1>
    </div>
  );
};

export { ProgressBar };
