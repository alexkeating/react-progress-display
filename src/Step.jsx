import React from 'react';
import PropTypes from 'prop-types';

const Step = ({ className }) => {
  return (
    <li className={className} >
      <span />
    </li>
  );
};

Step.propTypes = {
  className: PropTypes.string,
};

Step.defaultProps = {
  className: '',
};

export { Step };
