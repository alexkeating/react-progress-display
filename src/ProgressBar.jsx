/**
 * Created by alexkeating on 6/6/17.
 */

// What will this progress bar do?
// 0.1.0 should be a straight line that can be partitioned.
// Add styling to steps and to progressbar
// Add proptypes
// Step styling should be flexible
// Figure out mechanism to make everything active

import React from 'react';
import { Step } from './Step.jsx';

const ProgressBar = ({className, steps, activeSteps}) => {
  const numberOfSteps = Array.apply(null, {length: steps}).map(Number.call, Number);
  const listOfSteps = numberOfSteps.map(step => {
    return activeSteps.includes(step) ? <Step className='active' key={step} /> :
    <Step className='disabled' key={step} />;
  });

  return (
    <div className={className}>
      {listOfSteps}
    </div>
  );
};

export { ProgressBar };
