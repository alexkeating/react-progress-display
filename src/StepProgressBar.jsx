import React from 'react';
import PropTypes from 'prop-types';
import { Step } from './Step.jsx';

const StepProgressBar = ({className, steps, activeSteps}) => {
  const numberOfSteps = Array.apply(null, {length: steps}).map(Number.call, Number);
  const listOfSteps = numberOfSteps.map(step => {
    return activeSteps.includes(step) ? <Step className='progress-bar-active' key={step} /> :
    <Step className='progress-bar-disabled' key={step} />;
  });

  return (
    <div className={className}>
      {listOfSteps}
    </div>
  );
};

StepProgressBar.propTypes = {
  className: PropTypes.string,
  steps: PropTypes.number,
  activeSteps: PropTypes.arrayOf(PropTypes.number)
};

StepProgressBar.defaultProps = {
  className: '',
  steps: 0,
  activeSteps: [],
};

export { StepProgressBar };
