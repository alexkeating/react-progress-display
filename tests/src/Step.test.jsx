import React from 'react';
import { Step } from '../../src/Step';
import renderer from 'react-test-renderer';

test('Step active snapshot', () => {
  const component = renderer.create(
    <Step className='progress-bar-active' />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Step disabled snapshot', () => {
  const component = renderer.create(
    <Step className='progress-bar-disabled' />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
