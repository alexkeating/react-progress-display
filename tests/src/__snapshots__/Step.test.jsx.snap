// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step active snapshot 1`] = `
<li
  className="progress-bar-active"
>
  <span />
</li>
`;

exports[`Step disabled snapshot 1`] = `
<li
  className="progress-bar-disabled"
>
  <span />
</li>
`;
